### PR TITLE
Generate maps and sets

### LINKED ISSUE #1221: Generate elementMap with non-simplifiable keys for tests

### PR DESCRIPTION
**Pull request by virgil-serbanuta**
Fixes #1221 

---

###### Reviewer checklist

- [ ] Test coverage: `stack test --coverage`
- [ ] Public API documentation: `stack haddock`

### CODE REVIEW

File: kore/test/Test/Kore/Step/Simplification/IntegrationProperty.hs [hunk 1]
```diff
@@ -55,7 +55,7 @@ import Test.SMT
 
 test_simplifiesToSimplified :: TestTree
 test_simplifiesToSimplified =
-    testPropertyWithSolver "simplify returns simplified pattern" $ do
+    testPropertyWithSolver "zzzsimplify returns simplified pattern" $ do
```
Comment (ana-pantilie) [inline kore/test/Test/Kore/Step/Simplification/IntegrationProperty.hs]: Does the `zzz` mean anything?

Comment (virgil-serbanuta) [reply kore/test/Test/Kore/Step/Simplification/IntegrationProperty.hs]: Sorry, I use that to make it easier to run only one test.

File: kore/test/Test/ConsistentKore.hs [hunk 2]
```diff
@@ -610,41 +741,189 @@ maybeListBuiltinGenerator Setup { maybeListSorts } =
...
+                , attributes = AttributeRequirements
+                    { isNonSimplifiable = False
+                    -- We could generate non-simplifiable or concrete sets
+                    -- if we wanted to.
+                    , isConcrete = False
+                    }
+                , generator = setGenerator collectionSort elementSort
+                }
+  where
+    setGenerator
```
Comment (ttuegel) [inline kore/test/Test/ConsistentKore.hs]: This seems similar to the `mapGenerator` above; are they similar enough that we could refactor?

Comment (virgil-serbanuta) [reply kore/test/Test/ConsistentKore.hs]: Done.

Comment (ttuegel) [reply kore/test/Test/ConsistentKore.hs]: Looks great, thanks!

**Pull request merged**